tests/common/mod.rs

```
@@ -0,0 +1,76 @@
+/*
+ Copyright FMS Guardrails Orchestrator Authors
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+
+*/
+
+use std::sync::Arc;
+
+use fms_guardrails_orchestr8::config::OrchestratorConfig;
+use fms_guardrails_orchestr8::orchestrator::Orchestrator;
+use fms_guardrails_orchestr8::server::ServerState;
+use mocktail::generate_grpc_server;
+use mocktail::mock::MockSet;
+use mocktail::server::HttpMockServer;
+use rustls::crypto::ring;
+
+generate_grpc_server!(
+    "caikit.runtime.Chunkers.ChunkersService",
+    MockChunkersServiceServer
+);
+
+/// Default orchestrator configuration file for integration tests.
+pub const CONFIG_FILE_PATH: &str = "tests/test.config.yaml";
+
+///
+pub fn ensure_global_rustls_state() {
+    let _ = ring::default_provider().install_default();
+}
+
+/// Starts mock servers and adds them to orchestrator configuration.
+pub async fn create_orchestrator_shared_state(
+    detectors: Vec<HttpMockServer>,
+    chunkers: Vec<(&str, MockChunkersServiceServer)>,
+) -> Arc<ServerState> {
+    let mut config = OrchestratorConfig::load(CONFIG_FILE_PATH).await.unwrap();
+
+    for detector_mock_server in detectors {
+        let _ = detector_mock_server.start().await;
```

This would ignore errors during startup. We should use `mock_server.start().await?` and return `Result`

---

@declark1 Since this function won't exist anymore, is it okay to use this approach on test functions, or would it be cleaner to return a `Result`? I ask because tests are going to fail regardless.

---

Looks like #[tokio::test] doesn't allow returning `Result`, so `unwrap()` is needed at the test level.

---

Not accurate, you can definitely return `Result` when `#[tokio::test]` is used; I have definitely done this before.